db/migrate/20200402130013_add_repsoitory_docker_meta_tag_f_key.rb

```diff
@@ -17,7 +17,11 @@ def up
     add_index :katello_repository_docker_meta_tags, [:repository_id, :docker_meta_tag_id], :unique => true, :name => 'repository_docker_meta_tags_rid_dmtid'
 
     Katello::RepositoryDockerTag.where.not(:repository_id => Katello::Repository.select(:id)).delete_all
-    Katello::RepositoryDockerTag.where.not(:docker_tag_id => Katello::DockerTag.select(:id)).delete_all
+    Katello::RepositoryDockerTag
+      .joins("LEFT JOIN katello_docker_tags d on katello_repository_docker_tags.docker_tag_id = d.id")
+      .select("katello_repository_docker_tags.id")
+      .group("katello_repository_docker_tags.id")
+      .having("count(d.id) = 0").each { |t| t.delete }
```

Comment:
would it worth with .delete_all ?  I would think that'd be even faster

Reply:
`delete_all` doesn't support `group` and `having`, so I'll have to figure out a workaround.